compiler/luci/pass/src/RemoveNoEffectSlicePass.cpp

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/RemoveNoEffectSlicePass.h"
+
+#include <luci/IR/CircleNodes.h>
+
+namespace
+{
+
+template <loco::DataType DT>
+bool check_input_output_shape(luci::CircleNode *input, luci::CircleConst *begin,
+                              luci::CircleConst *size)
+{
+  for (uint32_t i = 0; i < input->rank(); i++)
+  {
+    int64_t size_value = static_cast<int64_t>(size->at<DT>(i));
+    if (size_value < 0)
+    {
+      if (size_value != -1)
+        return false;
+      size_value =
+          static_cast<int64_t>(input->dim(i).value()) - static_cast<int64_t>(begin->at<DT>(i));
+    }
+    else
+    {
+      if (input->shape_signature().dim(i) == -1)
+        return false;
+      if (static_cast<int64_t>(input->dim(i).value()) <
+          static_cast<int64_t>(begin->at<DT>(i)) + size_value)
+        return false;
```

Comment:
There is no check for `begin`. AFAIK, the value of `begin` must be 0 in all ranks.

If the check for begin is added, these lines will be unnecessary.

Reply:
Okay I will apply it.